prow/config/jobs/api-1.21.yaml

```diff
@@ -0,0 +1,782 @@
+branches:
+- release-1.21
+env:
+- name: BUILD_WITH_CONTAINER
+  value: "0"
+- name: BUILD_WITH_CONTAINER
+  value: "0"
+image: gcr.io/istio-testing/build-tools:release-1.21-19f0fb6c28d080531991023434dfadb2687444e2
+jobs:
+- command:
+  - make
+  - presubmit
+  env:
+  - name: BUILD_WITH_CONTAINER
+    value: "0"
+  - name: BUILD_WITH_CONTAINER
+    value: "0"
+  image: gcr.io/istio-testing/build-tools:master-19f0fb6c28d080531991023434dfadb2687444e2
```

Comment:
Also, you have many `image:` lines with `master` in them. With the exception if the proxy file, I think these can be mostly remove. Proxy has a global build-tools-proxy, but some jibs override it with the build-tools. Need to make sure this are done correctly.